packages/licenses/src/lockfileToLicenseNodeTree.ts

```diff
@@ -0,0 +1,161 @@
+> & {
+  install: string[];
+  remove: string[];
+  metadata: Object;
+};
+
+export interface LicenseExtractOptions {
+  virtualStoreDir: string;
+  modulesDir?: string;
+  dir: string;
+  getPackageInfo: GetPackageInfoFunction;
+}
+
+/**
+ * private
+ */
+export async function lockfileToLicenseNode(
+  step: LockfileWalkerStep,
+  options: LicenseExtractOptions
+) {
+  const dependencies = {};
+  for (const dependency of step.dependencies) {
+    const { depPath, pkgSnapshot, next } = dependency;
+    const { name, version } = nameVerFromPkgSnapshot(depPath, pkgSnapshot);
+
+    const modules = path.join(
+      options.dir,
+      options.virtualStoreDir,
+      dp.depPathToFilename(depPath),
+      options.modulesDir ?? "node_modules"
+    );
+
+    let packageDetails;
+    try {
+      packageDetails = await options.getPackageInfo({
+        name,
+        version,
+        prefix: modules,
+      });
+    } catch (err: unknown) {}
+
+    if (packageDetails) {
+      const subdeps = await lockfileToLicenseNode(next(), options);
+
+      const { packageInfo, packageManifest } = packageDetails;
+      const dep: LicenseNode = {
+        name,
+        dev: pkgSnapshot.dev === true,
+        integrity: pkgSnapshot.resolution["integrity"],
+        version,
+        packageManifest,
+        license: packageInfo.license,
+        licenseContents: packageInfo.licenseContents,
+        vendorName: packageInfo.author,
+        vendorUrl: packageInfo.homepage,
+        path: packageInfo.path,
+      };
+
+      if (Object.keys(subdeps).length > 0) {
+        dep.dependencies = subdeps;
+        dep.requires = toRequires(subdeps);
+      }
+
+      // If the package details could be fetched, we consider it part of the tree
+      dependencies[name] = dep;
+    }
+  }
+
+  return dependencies;
+}
+
+/**
+ * Reads the lockfile and converts it in a node tree of information necessary
+ * to generate the licenses summary
+ * @param lockfile the lockfile to process
+ * @param opts     parsing instructions
+ * @returns
+ */
+export async function lockfileToLicenseNodeTree(
+  lockfile: Lockfile,
+  opts: {
+    getPackageInfo: GetPackageInfoFunction;
+    include?: { [dependenciesField in DependenciesField]: boolean };
+  } & LicenseExtractOptions
+): Promise<LicenseNodeTree> {
+  const importerWalkers = lockfileWalkerGroupImporterSteps(
+    lockfile,
+    Object.keys(lockfile.importers),
+    { include: opts?.include }
+  );
+  const dependencies = {};
+
+  for (const importerWalker of importerWalkers) {
+    const importerDeps = await lockfileToLicenseNode(importerWalker.step, {
+      virtualStoreDir: opts.virtualStoreDir,
+      modulesDir: opts.modulesDir,
+      dir: opts.dir,
+      getPackageInfo: opts.getPackageInfo,
+    });
+
+    // For some reason the registry responds with 500 if the keys in dependencies have slashes
+    // see issue: https://github.com/pnpm/pnpm/issues/2848
```

Comment:
Isn't this comment for the audit command? Why is it needed here? The license command won't make a request to the audit endpoint.